### PR TITLE
Add documentation for inter-pod affinity/anti-affinity

### PR DESCRIPTION
**Pull request by davidopp**
I'm not sure why the code include lines aren't working; it seems to be following the directions in 
http://kubernetes.io/editdocs/
for how to include text from another file.

@kubernetes/docs to review and hopefully answer the question above

cc/ @kevin-wangzefeng just FYI

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1117)

### CODE REVIEW

File: docs/user-guide/node-selection/index.md [hunk 1]
```diff
@@ -1,16 +1,29 @@
 ---
 assignees:
-- dchen1107
-- eparis
-- mikedanese
+- davidopp
```
Comment (davidopp) [inline docs/user-guide/node-selection/index.md]: TODO: make this and the next line start with two dashes, not one

**Comment by janetkuo** [comment]
```
=== RUN   TestExampleObjectSchemas
--- FAIL: TestExampleObjectSchemas (0.03s)
	examples_test.go:293: ../docs/user-guide/node-selection/pod-with-pod-affinity.yaml: pod-with-pod-affinity does not have a test case defined
```

Add docs/user-guide/node-selection/pod-with-pod-affinity.yaml to the example tests in examples_test.go

**Comment by janetkuo** [comment]
@kevin-wangzefeng wanna take a look?

**Comment by davidopp** [comment]
I screwed up this PR pretty badly in my local client so have replaced it with #1148.

**Pull request closed without merging**